sktime/forecasting/arch.py

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+
+"""Implements Generalized Autoregressive Conditional Heteroskedasticity (GARCH) models and it's variants."""
+
+__author__ = ["Vasudeva-bit"]
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+_check_soft_dependencies("arch", severity="warning")
+
+class ARCH(BaseForecaster):
+    """Generalized Autoregressive Conditional Heteroskedasticity (GARCH) model for forecasting 
+    votility in high frequency time series data.
+
+    Implements Generalized Autoregressive Conditional Heteroskedasticity (GARCH) models introduced 
+    and implemented in [1]_, [2]_, and [3]_ to forecast the volatility in high frequency time series data.
+    
+    ARCH models are a popular class of volatility models that use observed values of returns or residuals 
+    as volatility shocks. A basic GARCH model is specified as
+        rt = μ+ϵt
```

Comment:
if you use math, kindly use the math environment in rst.
Also kindly ensure that all symbols are properly defined, if you use math, and clearly linked to the inputs y, X